questdb-rs/src/ingress/mod.rs

```diff
@@ -993,6 +995,121 @@ impl Buffer {
+    /// let array_2d = array![[1.1, 2.2], [3.3, 4.4]];
+    /// buffer.column_arr("array_col", &array_2d.view())?;
+    /// # Ok(())
+    /// # }
+    /// # }
+    ///
+    /// ```
+    ///
+    /// Using [`ColumnName`] for validated column names:
+    ///
+    /// ```
+    /// # #[cfg(feature = "ndarray")]
+    /// # {
+    /// # use questdb::Result;
+    /// # use questdb::ingress::{Buffer, ColumnName};
+    /// # use ndarray::Array3;
+    /// # fn main() -> Result<()> {
+    /// # let mut buffer = Buffer::new();
+    /// # buffer.table("x1")?;
+    /// // Record a 3D array of f64 values
+    /// let array_3d = Array3::from_elem((2, 3, 4), 42f64);
+    /// let col_name = ColumnName::new("col1")?;
+    /// buffer.column_arr(col_name, &array_3d.view())?;
+    /// # Ok(())
+    /// # }
+    /// # }
+    /// ```
+    /// # Errors
+    ///
+    /// Returns [`Error`] if:
+    /// - Array dimensions exceed [`MAX_DIMS`]
+    /// - Failed to get dimension sizes
+    /// - Memory allocation fails for array data
+    /// - Column name validation fails
+    pub fn column_arr<'a, N, T, D>(&mut self, name: N, view: &T) -> Result<&mut Self>
+    where
+        N: TryInto<ColumnName<'a>>,
+        T: NdArrayView<D>,
+        D: ArrayElement,
+        Error: From<N::Error>,
+    {
+        self.write_column_key(name)?;
+
+        // check dimension less equal than max dims
+        if MAX_DIMS < view.ndim() {
+            return Err(error::fmt!(
+                ArrayHasTooManyDims,
+                "Array dimension mismatch: expected at most {} dimensions, but got {}",
+                MAX_DIMS,
+                view.ndim()
+            ));
+        }
+
+        // binary format flag '='
+        self.output.push(b'=');
+        // binary format entity type
+        self.output.push(ARRAY_BINARY_FORMAT_TYPE);
+        // ndarr datatype
+        self.output.push(D::elem_type().into());
+        // ndarr dims
+        self.output.push(view.ndim() as u8);
+
+        let mut reserve_size = size_of::<D>();
+        for i in 0..view.ndim() {
+            let d = view.dim(i).ok_or_else(|| {
+                error::fmt!(
+                    ArrayViewError,
+                    "Can not get correct dimensions for dim {}",
+                    i
+                )
+            })?;
+
+            // ndarr shapes
+            self.output
+                .extend_from_slice((d as i32).to_le_bytes().as_slice());
+            reserve_size = reserve_size.checked_mul(d).ok_or(error::fmt!(
+                ArrayViewError,
+                "Array total elem size overflow"
+            ))?
+        }
+
+        let index = self.output.len();
+        let reserve_size = reserve_size.checked_add(index).ok_or(error::fmt!(
+            ArrayViewError,
+            "Array total elem size overflow"
+        ))?;
+        self.output
+            .try_reserve(reserve_size)
+            .map_err(|_| error::fmt!(BufferOutOfMemory, "Buffer out of memory"))?;
```

Comment:
That's not how reserve works: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.reserve

You pass in the argument for _additional_ required size, _not the total_.

Also `try_reserve` is completely pointless here since we can end up with large memory pressure already: A regular `reserve` should do.